mobius/src/geometry/integer_arcs.rs

```diff
@@ -0,0 +1,208 @@
+
+    Ok(Circle {
+        center: midpoint.into(),
+        radius,
+    })
+}
+
+pub fn arc_on_line_by_direction(
+    a: i64,
+    b: i64,
+    direction: ArcDirection,
+) -> Result<CircularArc, IntegerArcError> {
+    let circle = circle_on_line(a, b)?;
+
+    let angles = match direction {
+        ArcDirection::Counterclockwise if a < b => ArcAngles::new(PI, 2.0 * PI),
+        ArcDirection::Counterclockwise => ArcAngles::new(0.0, PI),
+        ArcDirection::Clockwise if a < b => ArcAngles::new(PI, 0.0),
+        ArcDirection::Clockwise => ArcAngles::new(0.0, -PI),
+    }?;
+
+    Ok(CircularArc { circle, angles })
+}
+
+#[derive(PartialEq)]
+pub enum Hemisphere {
+    North,
+    South,
+}
+
+pub fn arc_on_line_by_hemisphere(
+    a: i64,
+    b: i64,
+    hemisphere: Hemisphere,
+) -> Result<CircularArc, IntegerArcError> {
+    let circle = circle_on_line(a, b)?;
+
+    let angles = match hemisphere {
+        Hemisphere::North if a < b => ArcAngles::new(PI, 0.0),
+        Hemisphere::North => ArcAngles::new(0.0, PI),
+        Hemisphere::South if a < b => ArcAngles::new(PI, 2.0 * PI),
+        Hemisphere::South => ArcAngles::new(0.0, -PI),
+    }?;
+
+    Ok(CircularArc { circle, angles })
+}
+
+fn cyclotomic_angles(a: i64, b: i64, n: usize) -> Result<ArcAngles, IntegerArcError> {
+    if n == 0 {
+        return Err(IntegerArcError::ZeroPoints);
+    }
+
+    let n = n as i64;
+    if a >= n {
+        return Err(IntegerArcError::ValueOutOfRange("a".into(), a));
+    }
+
+    if b >= n {
+        return Err(IntegerArcError::ValueOutOfRange("b".into(), b));
+    }
+
+    if a == b {
+        return Err(IntegerArcError::DuplicateInt(a));
+    }
+    let a = a as f64;
+    let b = b as f64;
+
+    let step_size = TAU / (n as f64);
+    Ok(ArcAngles::new(a * step_size, b * step_size)?)
+}
+
+pub fn circle_on_circle(a: i64, b: i64, n: usize) -> Result<GeneralizedCircle, IntegerArcError> {
+    let angles = cyclotomic_angles(a, b, n)?;
+    Ok(compute_orthogonal_circle(Circle::unit_circle(), angles))
+}
+
+pub fn arc_on_circle_by_direction(
+    a: i64,
+    b: i64,
+    n: usize,
+    direction: ArcDirection,
+) -> Result<OrthogonalArc, IntegerArcError> {
+    let angles = cyclotomic_angles(a, b, n)?;
+    let arc = CircularArc::new(Circle::unit_circle(), angles);
+    let orthog_arc = compute_orthogonal_arc(arc);
+
+    let adjusted_arc = match orthog_arc {
+        OrthogonalArc::Arc(circular_arc) => {
+            let selected_arc = if circular_arc.direction() == direction {
+                circular_arc
+            } else {
+                circular_arc.complement()
+            };
+            OrthogonalArc::Arc(selected_arc)
+        }
+        OrthogonalArc::Diameter(line_segment) => {
+            // When do we flip this?
+            OrthogonalArc::Diameter(line_segment)
+        }
+        OrthogonalArc::DiameterOutside(_) => unreachable!(),
+    };
+
+    Ok(adjusted_arc)
+}
+
+pub fn arc_on_circle_by_hemisphere(
```

Comment:
Test cases:

n = 0 returns error
a or b out of range returns error
duplicate points return error
compute example for each of the 4 cases with n = 4
swapping a and b should produce the reverse
Check that opposite points with north hemisphere always produces a diameter outside
Check that opposite points with south hemisphere produces a diameter inside